app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeKeyActivity.java

```diff
@@ -184,7 +187,9 @@ public void onClick(final DialogInterface dialog, final int which) {
          * @param editName The textinput from which we show and get the name from.
          * @param key The key made from the QR code.
          */
-        private EditNameClickListener(final EditText editName, final IKey key) {
+        private EditNameClickListener(final Activity activity, final IServiceLocator serviceLocator, final EditText editName, final IKey key) {
```

Comment:
If I integrate it in the activity, it becomes a too-large anonymous class. In my opinion it's better to dependency inject stuff than to ask for it, see https://softwareengineering.stackexchange.com/questions/157526/explanation-on-how-tell-dont-ask-is-considered-good-oo